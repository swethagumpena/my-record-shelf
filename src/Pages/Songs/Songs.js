import React from 'react';
import PropTypes from 'prop-types';
import { useHistory, Link } from 'react-router-dom';
import Card from '../../components/Card/Card';
import styles from './Songs.module.scss';

const Songs = ({ songs }) => {
  const history = useHistory();
  return (
    <div>
      <div className={styles.header}>
        <p className={styles.heading}>all songs</p>
        <Link
          to="/genres"
          onClick={() => history.push('/genres')}
        >
          <img data-testid="toggle" className={styles.toggleImg} src="assets/icon-genre.svg" alt="toggle view" />
        </Link>
      </div>

      <div className={styles.songsContainer}>
        <div className={styles.card}>
          {songs.map((song) => (
            <Card
              key={song.id}
              id={song.id}
              albumArtUrl={song.albumArtUrl}
              songName={song.name}
              genreName={song.genre.name}
              artistName={song.artist.name}
            />
          ))}
        </div>
      </div>

    </div>
  );
};

// albumArtUrl: "https://i.scdn.co/image/ab67616d0000b273dc1f496d2a8d75f7fda092b6"
// artist: {id: "b22d7ed5-3ded-4958-a036-60453b7fc102", name: "Neha Kakkar"}
// genre: {id: "37146752-2c95-4204-a845-a32f3cdce9c8", name: "Bollywood"}
// id: "07075cdb-9e5c-41ec-8c09-9a1322dd92d8"
// name: "Aankh Marey (From "Simmba")"
// publishedAt: "2018-12-06T00:00:00"

const artistShape = PropTypes.shape({
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
});

const songShape = PropTypes.shape({
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  publishedAt: PropTypes.string.isRequired,
  albumArtUrl: PropTypes.string.isRequired,
  artist: PropTypes.objectOf(artistShape),
  genre: PropTypes.objectOf(artistShape),
});

Songs.propTypes = {
  songs: PropTypes.arrayOf(songShape).isRequired,
};
export default Songs;
