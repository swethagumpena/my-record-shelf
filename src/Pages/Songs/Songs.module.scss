@import "../../styles/styles.scss";

.songsContainer {
    display: flex;
}

.card {
    display: flex;
    flex-wrap: wrap;
    padding: 1rem;
    max-width: 100rem;
    padding: 0;
    justify-content: center;
}

.heading{
    padding: 2rem 2rem 2rem 4rem;
    font-family: 'Poppins', sans-serif;
    font-weight: bolder;
    font-size: 2rem;
}

p{
    margin: 0;
}