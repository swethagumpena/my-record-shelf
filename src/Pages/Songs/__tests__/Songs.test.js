/* eslint-disable max-len */
import { render, screen, fireEvent } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import React from 'react';
import Songs from '../Songs';

describe(Songs.name, () => {
  const mockSongs = [
    {
      albumArtUrl: 'https://i.scdn.co/image/ab67616d0000b273dc1f496d2a8d75f7fda092b6',
      artist:
        {
          id: 'b22d7ed5-3ded-4958-a036-60453b7fc102',
          name: 'Neha Kakkar',
        },
      genre:
        {
          id: '37146752-2c95-4204-a845-a32f3cdce9c8',
          name: 'Bollywood',
        },
      id: '07075cdb-9e5c-41ec-8c09-9a1322dd92d8',
      name: 'Aankh Marey (From Simmba)',
      publishedAt: '2018-12-06T00:00:00',
    },
  ];

  afterEach(() => {
    jest.clearAllMocks();
  });

  test('should match snapshot', () => {
    const { container } = render(<BrowserRouter><Songs songs={mockSongs} /></BrowserRouter>);
    expect(container).toMatchSnapshot();
  });

  test('should display all songs', () => {
    render(<BrowserRouter><Songs songs={mockSongs} /></BrowserRouter>);
    expect(screen.getByText('all songs'));
  });

  test('should display a button to toggle view', () => {
    render(<BrowserRouter><Songs songs={mockSongs} /></BrowserRouter>);
    const songElement = screen.getByTestId('toggle');
    fireEvent.click(songElement);
    expect(document.location.href).toBe('http://localhost/genres');
  });
});
