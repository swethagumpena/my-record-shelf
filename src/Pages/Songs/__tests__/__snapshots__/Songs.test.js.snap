// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Songs should match snapshot 1`] = `
<div>
  <div>
    <div
      class="header"
    >
      <p
        class="heading"
      >
        all songs
      </p>
      <a
        href="/genres"
      >
        <img
          alt="toggle view"
          class="toggleImg"
          data-testid="toggle"
          src="assets/icon-genre.svg"
        />
      </a>
    </div>
    <div
      class="songsContainer"
    >
      <div
        class="card"
      >
        <div
          class="card"
        >
          <img
            alt="song"
            data-testid="song-image"
            src="https://i.scdn.co/image/ab67616d0000b273dc1f496d2a8d75f7fda092b6"
          />
          <div
            class="content"
          >
            <div
              class="text"
            >
              <p
                class="song"
              >
                Aankh Marey (From Simmba)
              </p>
              <p
                class="artist"
              >
                Neha Kakkar
              </p>
            </div>
            <div
              class="likeButton"
            >
              <button
                type="button"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
